src/Beacon.jsx

```diff
@@ -77,8 +82,28 @@ export class Beacon extends Component {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextState.tooltip) {
+      // Always update if the tooltip is visible
+      return true;
+    } else if (Object.keys(this.props).some(key => this.props[key] !== nextProps[key])) {
```

Comment:
You know that this is something you never want to write on your own? https://github.com/gaearon/react-pure-render/blob/master/src/shallowEqual.js

I assume that the extra logic is probably mandatory.